sys-cluster/teleport/teleport-2.4.0.ebuild

```diff
@@ -0,0 +1,51 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+inherit eutils golang-build systemd user
+
+DESCRIPTION="Modern SSH server for teams managing distributed infrastructure"
+HOMEPAGE="https://gravitational.com/teleport"
+
+EGO_PN="github.com/gravitational/${PN}/..."
+
+if [[ ${PV} == "9999" ]] ; then
+	inherit git-r3 golang-vcs
+	EGIT_REPO_URI="https://github.com/gravitational/${PN}.git"
+else
+	inherit golang-vcs-snapshot
+	SRC_URI="https://github.com/gravitational/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~arm"
+fi
+
+LICENSE="Apache-2.0"
+SLOT="0"
+IUSE=""
+RESTRICT="test strip"
+
+DEPEND="
+	app-arch/zip
+	>=dev-lang/go-1.8.3"
+RDEPEND=""
+
+src_compile() {
+	GOPATH="${S}" emake -j1 -C src/${EGO_PN%/*} full
+}
+
+src_install() {
+	dodir /var/lib/${PN} /etc/${PN}
```

Comment:
Why are you doing this? `dodir` is generally to be used only when you need to create a directory into which you're (manually) installing a file, i.e. as replacement for pure `mkdir`. If you need the directory kept, you need `keepdir`. Otherwise it may be stripped as an empty directory.